tapir/shifts/templatetags/shifts.py

```diff
@@ -122,13 +135,11 @@ def shift_template_to_block_object(shift_template: ShiftTemplate, fill_parent: b
 
         attendances[slot_name].append(state)
 
-    required_slot_templates = shift_template.slot_templates.filter(optional=False)
-    num_required_slots = required_slot_templates.count() or 1
+    num_required_slots = len(
```

Comment:
I would personally prefer a functional style here, something like:

```
required_slots = shift_template.slot_templates.filter(optional=False)
num_required_slots = required_slots.count()
num_attendances_in_required_slots = len([_ for slot in required_slots if slot.get_attendance_template() is not None])
```

Reply:
Both those changes are here to reduce DB queries. I used a lot of prefetch_related in the view to preload the related objects but if we use any filter it does query again. It would be fine on most views but since this gets repeated for each shift in the overview, so the number of queries quickly goes crazy.